core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -266,14 +257,40 @@ public void commitBlock(long sessionId, long blockId, boolean pinOnCreate)
       throws BlockAlreadyExistsException, InvalidWorkerStateException, BlockDoesNotExistException,
       IOException {
     LOG.debug("commitBlock: sessionId={}, blockId={}", sessionId, blockId);
-    BlockStoreLocation loc = commitBlockInternal(sessionId, blockId, pinOnCreate);
-    synchronized (mBlockStoreEventListeners) {
-      for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
-        listener.onCommitBlock(sessionId, blockId, loc);
+    long lockId = mLockManager.lockBlock(sessionId, blockId, BlockLockType.WRITE);
```

Comment:
can `commitBlock` reuse `commitBlockLocked`?

Reply:
No. The purpose of this API is to retain the lock after returning from this. They both use `commitBlockInternal` though